src/front/glsl/variables.rs

```diff
@@ -0,0 +1,81 @@
+use crate::{
+    Binding, BuiltIn, Expression, GlobalVariable, Handle, ScalarKind, ShaderStage, StorageAccess,
+    StorageClass, Type, TypeInner, VectorSize,
+};
+
+use super::ast::*;
+
+pub fn lookup_variable(program: &mut Program, name: &str) -> Option<Handle<Expression>> {
+    let mut expression: Option<Handle<Expression>> = None;
+    if name == "gl_Position" {
+        if let ShaderStage::Vertex | ShaderStage::Fragment { .. } = program.shader_stage {
+            let h = program.global_variables.fetch_or_append(GlobalVariable {
+                name: Some(name.into()),
+                class: match program.shader_stage {
+                    ShaderStage::Vertex => StorageClass::Output,
+                    ShaderStage::Fragment { .. } => StorageClass::Input,
+                    _ => StorageClass::Input,
```

Comment:
what is this? compute? we should generate an error here

Reply:
I don't think we should error, in compute you could still make your own var named 'gl_Position' (maybe not very sensible, but you coudl). Anyway we are already checking on the stage in outer if, so I guess this is only to satisfy that match must cover all variants, but maybe there is a better way?

Reply:
if it's already checked, then it's not expected here, and thus it should be `unreachable!()`